textile/src/main/java/io/textile/textile/Textile.java

```diff
@@ -178,17 +168,83 @@
+     */
+    public static MobileWalletAccount walletAccountAt(final String phrase, final long index, final String password) throws Exception {
+        final byte[] bytes = Mobile.walletAccountAt(phrase, index, password);
+        return MobileWalletAccount.parseFrom(bytes);
+    }
+
+    /**
+     * Check if Textile is already initialized
+     * @param repoPath The path to the Textile repo
+     * @return A boolean value indicating if Textile is initialized or not
+     */
+    public static Boolean isInitialized(final String repoPath) {
+        File file = new File(repoPath);
+        return file.exists() && file.isDirectory();
+    }
+
+    /**
+     * Initialize the shared Textile instance with an existing account seed
+     * @param repoPath The path to the Textile repo
+     * @param seed The account seed
      * @param debug Sets the log level to debug or not
      * @param logToDisk Whether or not to write Textile logs to disk
-     * @return The wallet recovery phrase if it's the first time the node is being initialize, nil otherwise
      * @throws Exception The exception that occurred
      */
-    public static String initialize(final Context applicationContext, final boolean debug, final boolean logToDisk) throws Exception {
-        if (Textile.instance().node != null) {
-            return null;
-        }
+    public static void initialize(final String repoPath, final String seed, final boolean debug, final boolean logToDisk) throws Exception {
+        final InitConfig config = new InitConfig();
+        config.setSeed(seed);
+        config.setRepoPath(repoPath);
+        config.setLogToDisk(logToDisk);
+        config.setDebug(debug);
+        Mobile.initRepo(config);
+    }
+
+    /**
+     * Initialize the shared Textile instance, creating a new wallet
+     * @param repoPath The path to the Textile repo
+     * @param debug Sets the log level to debug or not
+     * @param logToDisk Whether or not to write Textile logs to disk
+     * @return The wallet recovery phrase
+     * @throws Exception The exception that occurred
+     */
+    public static String initializeCreatingNewWalletAndAccount(final String repoPath, final boolean debug, final boolean logToDisk) throws Exception {
+        final String recoveryPhrase = Textile.newWallet(WALLET_WORD_COUNT);
+        final MobileWalletAccount account = Textile.walletAccountAt(recoveryPhrase, WALLET_ACCOUNT_INDEX, WALLET_PASSPHRASE);
+        Textile.initialize(repoPath, account.getSeed(), debug, logToDisk);
+        return recoveryPhrase;
+    }
+
+    /**
+     * After initialization is complete, launch Textile
+     * @param applicationContext The application context
+     * @param repoPath The path to the Textile repo
+     * @param debug Sets the log level to debug or not
+     * @throws Exception The exception that occurred
+     */
+    public static void launch(final Context applicationContext, final String repoPath, final boolean debug) throws Exception {
```

Comment:
After things are initialized, `launch` sets up all the needed state and kicks things off.